service/db.go

```diff
@@ -462,58 +462,67 @@ func updateUserRow(uuid string, svcDerived *pblib.User, dbDerived *pblib.User) (
 	return updatedUser, nil
 }
 
-//TODO work on later
-func insertNewSecret(secret *pblib.Secret) error {
-	if secret == nil {
-		return authconst.ErrNilSecret
+// insertNewSecret inserts a newly generated secret key to database
+// secret key is used to sign JWT's
+// Returns err if secret is empty or error with database
+func insertNewSecret() error {
+	// generate a new secret
+	secretKey, err := generateToken(auth.SecretByteSize)
+	if err != nil {
+		return err
 	}
 
 	command := `
 				INSERT INTO user_security.secret(
-					secret_key, created_timestamp
-				) VALUES($1, $2)
+					secret_key, created_timestamp, expiration_timestamp
+				) VALUES($1, $2, $3)
 				`
 
-	_, err := postgresDB.Exec(command, secret.Key, secret.CreatedTimestamp)
-
+	expirationTimestamp, err := generateSecretExpirationDate(time.Now().UTC())
```

Comment:
The name of `generateSecretExpirationDate` should be `generateSecretExpirationTimestamp` because you are not generating a date but a timestamp.